### PR TITLE
closes #12441: Refactor VEHICLE_TRACK_SUBPOSITION to use strong enum

### LINKED ISSUE #12441: Refactor VEHICLE_TRACK_SUBPOSITION to use strong enum

### PR DESCRIPTION
**Pull request by autinerd**
This closes #12441: Refactor VEHICLE_TRACK_SUBPOSITION to use strong enum

### CODE REVIEW

File: src/openrct2/rct1/S4Importer.cpp [hunk 1]
```diff
@@ -1224,7 +1224,7 @@ class S4Importer final : public IParkImporter
             statusSrc = static_cast<VEHICLE_STATUS>(src->status);
         }
         dst->status = statusSrc;
-        dst->TrackSubposition = src->TrackSubposition;
+        dst->TrackSubposition = (VehicleTrackSubposition)src->TrackSubposition;
```
Comment (duncanspumpkin) [inline src/openrct2/rct1/S4Importer.cpp:1227]: You can construct it instead `VehicleTrackSubposition{src->TrackSubposition}`

File: src/openrct2/ride/Vehicle.h [hunk 2]
```diff
@@ -152,30 +152,30 @@ enum VEHICLE_STATUS
 
 struct rct_vehicle_sound_params;
 
-enum VEHICLE_TRACK_SUBPOSITION : uint8_t
+enum class VehicleTrackSubposition : uint8_t
 {
-    VEHICLE_TRACK_SUBPOSITION_0,
+    None,
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.h:157]: `Default` as its not nothing its just what all normal rides are

**Comment by duncanspumpkin** [comment]
Use static_cast for the conversions to `uint8_t`. You can use the constructor for conversions from `uint8_t` to your enum class. This comment applies for all your PR's.

File: src/openrct2/ride/Vehicle.h (hunk 2 above)
Comment (autinerd) [reply src/openrct2/ride/Vehicle.h:157]: Okay, the 0 didn't give much information, I will change it.

**Pull request merged**